src/features/Admin/Facilities/Facilities.tsx

```diff
@@ -1,19 +1,23 @@
 import axios from "axios";
 import { useContext, useEffect, useState } from "react";
 import { AuthContext } from "../../../context/AuthContext";
-import { facilitiesUrl } from "../../../services/api";
+import { facilitiesRoomsUrl } from "../../../services/api";
 import CustomTable from "../../UI/CustomTable/CustomTable";
-import Button from "@mui/material/Button";
-import CustomDialog from "./../../UI/CustomDialog/CustomDialog";
-import TextField from "@mui/material/TextField";
-import { SubmitHandler, useForm } from "react-hook-form";
-import { Box , Grid} from "@mui/material";
+import styleFacilities from "./Facilities.module.scss";
 
+import {
+  AppBar,
+  Button,
+  Grid,
+  TextField,
+  Typography
+} from "@mui/material";
+import { SubmitHandler, useForm } from "react-hook-form";
+import noData from '../../../assets/images/noData.png';
+import CustomButton from "./../../UI/CustomButton/CustomButton";
+import CustomModal from "./../../UI/CustomModal/CustomModal";
 
 const Facilities = () => {
```

Comment:
Consider breaking down your component into smaller components. For example, you can have separate components for Add Modal, View Modal, Update Modal, and Delete Modal. This makes your code more modular and easier to understand and more concise.